src/content/keyAutoAdd/lib/keyAutoAdd.ts

```diff
@@ -121,24 +121,19 @@ export class KeyAutoAdd {
     return promise;
   }
 
-  private async run({
-    walletAddressUrl,
-    publicKey,
-    nickName,
-    keyId,
-    keyAddUrl,
-  }: BeginPayload) {
-    const params: StepRunParams = {
-      walletAddressUrl,
-      publicKey,
-      nickName,
-      keyId,
-      keyAddUrl,
+  private async run(payload: BeginPayload) {
+    const helpers: StepRunHelpers = {
+      output: <T extends StepRun>(fn: T) => {
+        if (!this.outputs.has(fn)) {
```

Comment:
Just for my understanding, all steps will have an output?

Reply:
- All steps will have output, unless they throw (error or skip).
- If they return nothing, output is `undefined`, but still exists.
- Only reason `this.outputs.has(fn)` can be false is when the step hasn't run yet, or was skipped/errored.

Reply:
Can't we populate the outputs from the beginning and have `undefined` for no output yet and something else if they return nothing - like an empty object?

Reply:
So, the way `output(fn)` is supposed to be called, is, it has to make sure that output you're trying to get infact exists. For example, `waitForLogin` must not call `output(addKey)`. This throwing will ensure we catch such bugs in dev only.

If they return nothing, `output(fn)` will return `undefined` (and not throw).